tests/php/Http/Controllers/Api/VolumeControllerTest.php

```diff
@@ -198,17 +235,420 @@ public function testUpdateValidation()
+        ]);
+        ImageAnnotationLabelTest::create([
+            'annotation_id' => $a->id,
+            'user_id' => $this->admin()->id,
+        ]);
+        $a2 = ImageAnnotationTest::create([
+            'image_id' => $img1->id,
+            'created_at' => '2015-03-03',
+        ]);
+        ImageAnnotationLabelTest::create([
+            'annotation_id' => $a2->id,
+            'user_id' => $this->admin()->id,
+        ]);
+
+        $img2 = ImageTest::create([
+            'lng' => 9.5,
+            'lat' => 9.0,
+            'filename' => 'b.jpg',
+            'volume_id' => $this->volume->id,
+        ]);
+        ImageLabelTest::create(['image_id' => $img2->id]);
+
+        $img3 = ImageTest::create([
+            'filename' => 'c.jpg',
+            'volume_id' => $this->volume->id,
+        ]);
+        $a3 = ImageAnnotationTest::create([
+            'image_id' => $img3->id,
+            'created_at' => '2000-09-10',
+        ]);
+        ImageLabelTest::create(['image_id' => $img2->id]);
+        ImageLabelTest::create(['image_id' => $img2->id]);
+
+        $a3 = ImageAnnotationTest::create([
+            'image_id' => $img3->id,
+            'created_at' => '2000-09-10',
+        ]);
+        ImageAnnotationLabelTest::create([
+            'annotation_id' => $a3->id,
+            'user_id' => $this->admin()->id,
+        ]);
+
+        $session = AnnotationSessionTest::create([
+            'volume_id' => $id,
+            'starts_at' => Carbon::today(),
+            'ends_at' => Carbon::tomorrow(),
+            'hide_own_annotations' => true,
+            'hide_other_users_annotations' => false,
+        ]);
+        $u1 = UserTest::create();
+        $u2 = UserTest::create();
+        $u3 = UserTest::create();
+        $session->users()->attach($this->admin()->id);
+        $session->users()->attach($u1);
+        $session->users()->attach($u2);
+        $session->users()->attach($u3);
+
+
+        $project->addUserId($this->admin()->id, Role::adminId());
+        Cache::flush();
+
+        $response = $this->post("/api/v1/volumes/{$id}/project/{$id2}");
+        $response->assertStatus(302);
+
+
+        $copy = $response->getSession()->get('copy');
+
+        $this->assertTrue($copy->images()->exists() == $this->volume->images()->exists());
+        $this->assertTrue($copy->videos()->exists() == $this->volume->videos()->exists());
+
+        foreach ($this->volume->images()->get() as $index => $oldImage) {
+            $newImage = $copy->images()->get()[$index];
+            $this->assertTrue($oldImage->volume_id == $this->volume->id);
+            $this->assertTrue($newImage->volume_id == $copy->id);
+            unset($oldImage->volume_id);
+            unset($oldImage->id);
+            unset($oldImage->uuid);
+            unset($newImage->volume_id);
+            unset($newImage->id);
+            unset($newImage->uuid);
+            $this->assertEquals($newImage->getAttributes(), $oldImage->getAttributes());
```

Comment:
The first argument of `assertEquals()` should be the expected value and the second argument should be the value to check. Otherwise the error message is misleading if an assertion fails.